### ISSUE TITLE
Submitting a 500k bytes PFD causes the chain to get stuck in rounds 

### ISSUE THREAD

**Issue by Bidon15** [celestiaorg/celestia-app#563]
### Summary of Bug

When submitting a PFD with 500k bytes, the chain get stuck in rounds and can not proceed with generating a new block on a timely manner (test-case falls to fail after aprox 4 min of wait time for new block) 

Ref: https://github.com/celestiaorg/test-infra/pull/48
Logs: https://gist.github.com/Bidon15/56997a852e8cce67768be50c138e2a7c

### Version

v0.6.0

### config.toml for each of the validators
```bash
mempool.max-txs-bytes = 1073741824
mempool.max-tx-bytes = 1048576
mempool.size = 5000
consensus.timeout-commit = 30s
```

#### Note
If only 1 of 3 validators submit PFD containing 500k bytes of data, then it takes aprox. 142 seconds and 9 rounds to produce a block 
More info here: https://gist.github.com/Bidon15/008e7b2952a363c3a117649617ee3d9c
Some logs to search
```bash
Jul 26 14:52:52.968538	INFO	62.3979s    MESSAGE << validators[001] (c6e79b) >> Submitting PFD with 500k bytes random data
Jul 26 14:55:39.132887	INFO	228.5622s    MESSAGE << validators[001] (c6e79b) >> latest size of the block is - 501173
Jul 26 14:55:35.893631	INFO	225.3235s      OTHER << validators[002] (d28ccb) >> 2:55PM INF committed state app_hash=3EB1A85C2A11CDACE719A6B74A22FDE89425C64AFE373578FE5A500C171DCD8C height=3 module=state num_txs=1
```

### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by Bidon15** [comment]
I'll be updating with the new data on validators set quantity(let's say 30) vs amount of 500k txs (starting with 1tx in block and moving forward)
Behaviour is pretty much similar to what we see on mamaki

As well as this needs to be retested once we downgrade to 0.34.x tendermint

**Comment by evan-forbes** [comment]
> Behaviour is pretty much similar to what we see on mamaki

can you elaborate on this?

**Comment by evan-forbes** [comment]
I've tested this on mamaki, and large blocks do seem to increase the time it takes to produce a block, but the results vary. Note that this also seems to occur with other tx types besides PayForData.

**Comment by Bidon15** [comment]
> can you elaborate on this? 

Sure. I do remember that during the start of the mamaki chain, we had very big delays for creating new blocks (i don't know have we found the root cause and fix it) and not seeing any panics messages. And block times we had was like 4-5 min blocks

I've caught some of it again after playing around with pfd on mamaki (7 minutes block between 150940 to 150941)
<img width="1496" alt="Screenshot 2022-07-27 at 16 48 26" src="https://user-images.githubusercontent.com/1822365/181277954-4ab63c74-e1e7-4d22-83f0-e0286295227d.png">

some of the logs on 150940 block
https://gist.github.com/Bidon15/9fecfb69ea183770fe261884ddddc1e9

and .json output of the block rest request/response
https://gist.github.com/Bidon15/fd9a7863ec7735c4d1730a671431578d

**Comment by liamsi** [comment]
We should re-test this once the downgrade to tendermint 0.34 is complete. (re #472)

**Comment by Bidon15** [comment]
Results on downgraded tendermint and celestia-app cosmos sdk(both sdk/app are on my forks) based on 
https://github.com/celestiaorg/celestia-app/pull/472
https://github.com/celestiaorg/cosmos-sdk/pull/243

### Network setup: 
3 validators
`consensus.timeout_commit` = 60 seconds

#### Case 1️⃣ : All 3 validators submits
Outcome -> 2/3 have error : `RPC error -32603 - Internal error: timed out waiting for tx to be included in a block`
https://gist.github.com/Bidon15/16e6fac365c63f7c8f323d2a4d2f650e

#### Case 2️⃣  : One validator submits pfd with block flag
Outcome ->  Validator error : `RPC error -32603 - Internal error: timed out waiting for tx to be included in a block`
https://gist.github.com/Bidon15/63ad39d35c0b8033eb728a9b6c08cf75

#### Case 3️⃣  : One validator submits pfd with sync flag
Outcome -> No data was included
https://gist.github.com/Bidon15/3b5a0afaf9a3945b079421ef0137c9c1

**Comment by Bidon15** [comment]
This issue can be closed after we finish testing on the cluster with 15+ validators set and 4MB blocks
So far, local docker testing have shown that we can produce 1.5 MB blocks constantly having only 3 validators (which is the original for creating this issue) 

Proof: https://gist.github.com/Bidon15/5a906b597cc0075ff89f639e11004ba4

**Comment by Bidon15** [comment]
Closing as with 20 validators' set and 4mb blocks - we are safe and sound on producing blocks. 
The different issue is the bandwidth to withstand the network load, which is out-of-scope of this issue
https://gist.github.com/Bidon15/276f5abf7e102d67c2ac967d904bb5d2

**Comment by liamsi** [comment]
So the downgrade fixed this?

**Comment by Bidon15** [comment]
> So the downgrade fixed this?

Yeap!

**Comment by liamsi** [comment]
Can you let the tendermint team now via an issue as discussed, please?
